src/databricks/labs/lsql/dashboards.py

```diff
@@ -706,21 +709,33 @@ def replace_database(self, *args, **kwargs) -> "DashboardMetadata":
                 tiles.append(tile)
         return dataclasses.replace(self, _tiles=tiles)
 
-    def get_datasets(self) -> list[Dataset]:
+    def _get_datasets(self) -> list[Dataset]:
         """Get the datasets for the dashboard."""
         datasets: list[Dataset] = []
         for tile in self.tiles:
             if isinstance(tile, QueryTile):
                 datasets.extend(tile.get_datasets())
         return datasets
 
-    def get_layouts(self) -> list[Layout]:
+    def _get_layouts(self) -> list[Layout]:
         """Get the layouts for the dashboard."""
         layouts: list[Layout] = []
         for tile in self.tiles:
             layouts.extend(tile.get_layouts())
         return layouts
 
+    def as_lakeview(self) -> Dashboard:
+        """Create a lakeview dashboard from the dashboard metadata."""
+        datasets = self._get_datasets()
```

Comment:
@nfx : My intuition is to remove the `validate` now as it is a side effect of `as_lakeview` and should be called by the user before `as_lakeview`. We also do not `validate` in the `as_dict`